### ISSUE TITLE
harness callListener fails when the return of a render is a DNode[]

### ISSUE THREAD

**Issue by kitsonk** [dojo/test-extras#87]
**Bug** 

When a widget renders a `DNode[]` instead of a `DNode`, the `callListener()` method (and I suspect other methods) cannot resolve a target, because it is expecting a root `DNode` instead of an array.

**Code**

```ts
class TestWidget extends WidgetBase {
  render() {
    return [ v('div', { onclick() { console.log('hello'); }, key: 'foo' }) ];
  }
}

const widget = harness(TestWidget);
widget.callListener('onclick', { key: 'foo' });
```

**Expected behavior:**

The `onclick` is called.

**Actual behavior:**

```
TypeError: Cannot read property 'key' of undefined
  at findByKey  <node_modules/@dojo/test-extras/support/d.ts:126:23>
  at Object.findKey  <node_modules/@dojo/test-extras/support/d.ts:177:10>
  at Object.callListener [as default]  <node_modules/@dojo/test-extras/support/callListener.ts:48:20>
  at Harness.callListener  <node_modules/@dojo/test-extras/harness.ts:291:15>
```

**Comment by edhager** [comment]
I fixed this as part of PR #90.

I created unit tests that should be testing this scenario.

**Comment by edhager** [comment]
Fixed in PR #90.
